datamonster_api/lib/datamonster.py

```diff
@@ -247,3 +256,219 @@ def _avro_to_df(self, avro_buffer):
+                result['split_combination']['section_pk"]
+            is a list of section_pk's (though we acommodate a single pk or None);
+            we replace each pk with
+                self._pk_to_ticker(pk)          if ticker is not None,
+                with str(pk) + '-NO_TICKER'     if ticker is None
+        """
+        combo = dimension['split_combination']
+        if 'section_pk' in combo:
+            value = combo.pop('section_pk')
+            if value is not None:
+                combo['ticker'] = (
+                    self._pk_to_ticker(value, pk2ticker_memos)
+                    if isinstance(value, int) else          # isinstance(value, list) -- List[int] in fact
+                    six.moves.map(lambda pk: self._pk_to_ticker(pk, pk2ticker_memos), value)
+                )
+        return dimension
+
+    def _pk_to_ticker(self, pk, pk2ticker_memos):
+        """
+        :param pk: int -- a section_pk
+        :param pk2ticker_memos: (dict) maps section_pk => ticker
+            section_pk's already in this dict will use the tickers already looked up and saved;
+            newly-encountered section_pk's will have their tickers saved here
+
+        :return: str --
+                dm.get_company_from_pk(pk).ticker   if that is not None,
+                str(pk) + '-NO_TICKER'              otherwise (actual ticker is None or empty)
+        """
+        if pk not in pk2ticker_memos:
+            ticker = self.get_company_by_pk(pk).ticker
+            if not ticker:
+                ticker = str(pk) + '-NO_TICKER'
+            pk2ticker_memos[pk] = ticker
+
+        return pk2ticker_memos[pk]
+
+    @staticmethod
+    def to_json_checked(filters):
+        """
+        Not "private" because Datasource.get_dimensions() uses it too
+
+        :param filters: dict
+        :return: JSON string encoding `filters`. Normal exit if `filters` is
+            JSON-serializable.
+
+        :raises: DataMonsterError if `filters` isn't a dict or can't be JSON-encoded.
+        """
+        if not isinstance(filters, dict):
+            raise DataMonsterError(
+                "`filters` must be a dict, got {} instead".format(type(filters).__name__)
+            )
+        try:
+            return json.dumps(filters)
+        except TypeError as e:
+            raise DataMonsterError(
+                "Problem with filters when getting dimensions: {}".format(e)
+            )
+
+    def _get_dimensions_path(self, uuid):
+        return self.dimensions_path.format(uuid)
+
+
+class DimensionSet(object):
+    def __init__(self, url, dm, _pk2ticker):
+        """
+        :param url: (string) URL for REST endpoint
+        :param dm: DataMonster object
+        :param _pk2ticker: (bool) If True, convert 'section_pk' items to 'tickers' items
+        """
+        self._url_orig = url
+
+        resp0 = dm.client.get(url)
+
+        self._min_date = resp0['min_date']
```

Comment:
why do we make these private?

Reply:
Because there are corresponding public read-only properties.